src/common/key_value_database/platform/lmdb/lmdb.cpp

```diff
@@ -90,23 +89,23 @@ KeyValueDatabaseLmdb::~KeyValueDatabaseLmdb() {
 error::Error KeyValueDatabaseLmdb::Open(const string &path) {
 	Close();
 
+	env_ = make_unique<lmdb::env>(lmdb::env::create());
+
 	try {
 		env_->open(path.c_str(), MDB_NOSUBDIR, 0600);
 	} catch (std::runtime_error &e) {
+		env_.reset();
 		return MakeError(LmdbError, e.what());
 	}
 
-	successfully_opened_ = true;
 	return error::NoError;
 }
 
 void KeyValueDatabaseLmdb::Close() {
-	if (!successfully_opened_) {
-		return;
+	if (env_) {
+		env_->close();
```

Comment:
Q: Is the env not closed when it goes out of scope?

Reply:
Oh, you're right, this is redundant now.